ghcide/session-loader/Development/IDE/Session.hs

```diff
@@ -247,7 +247,7 @@ loadSessionWithOptions SessionLoadingOptions{..} dir = do
                       } <- getShakeExtras
     let invalidateShakeCache = do
             void $ modifyVar' version succ
-            recordDirtyKeys extras GhcSessionIO [emptyFilePath]
+            atomically $ recordDirtyKeys extras GhcSessionIO [emptyFilePath]
```

Comment:
I just noticed, this changes the type of `invalidateShakeCache` to `:: IO (IO ())`. I suppose that is unintentional?

Reply:
I think, either here we need a `join` or have to consume the action produced at: https://github.com/haskell/haskell-language-server/pull/2434/files#diff-d7140bbfca1b7aaafd8c82d1d6aa21a3f941eb27ba8f08e4d9fb13766b21b418L393